admiral/pkg/clusters/serviceentry.go

```diff
@@ -759,7 +759,7 @@ func modifyServiceEntryForNewServiceOrPod(
 			remoteRegistry.AdmiralCache.DependencyNamespaceCache.Put(val, serviceInstance[appType[sourceCluster]].Namespace, localFqdn, cnames)
 		}
 
-		if common.IsVSBasedRoutingEnabled() {
+		if common.DoVSRoutingForCluster(sourceCluster) {
```

Comment:
Is this discover phase, could we rename to `DiscoverVS(sourceCluster)`?

Reply:
@adilfulara , this just a feature flag for enabling/disabling VS based routing